cron-jobs/removeUnverifiedUsers.js

```diff
@@ -0,0 +1,31 @@
+const CronJob = require('cron').CronJob
+const Token = require('~/models/token')
+const tokenService = require('~/services/token')
+const userService = require('~/services/user')
+
+const EVERY_MIDNIGHT = '0 0 * * *'
+
+const removeUnverifiedUsers = new CronJob(EVERY_MIDNIGHT, () => removeUsersWithUnconfirmedEmail())
+
+const removeUsersWithUnconfirmedEmail = async () => {
+  const usersWithConfirmToken = await Token.find({ confirmToken: { $ne: null } })
+    .populate('user')
+    .lean()
+    .exec()
+  const unconfirmedUsersData = usersWithConfirmToken.filter(({ user }) => user && !user.isEmailConfirmed)
+
+  return Promise.all(
```

Comment:
I think you should add `await` here, for the possibly rejected promise to be caught in our `catch` block.